gmt-4.5.14/src/psbasemap.1
.TH PSBASEMAP 1 "1 Nov 2015" "GMT 4.5.14" "Generic Mapping Tools"
.SH NAME
psbasemap \- To plot \fIPostScript\fP basemaps
.SH SYNOPSIS
\fBpsbasemap\fP \fB\-B\fP[\fBp\fP|\fBs\fP]\fIparameters\fP \fB\-J\fP\fIparameters\fP \fB\-R\fP\fIwest\fP/\fIeast\fP/\fIsouth\fP/\fInorth\fP[/\fIzmin\fP/\fIzmax\fP][\fBr\fP] 
[ \fB\-E\fP\fIazim\fP/\fIelev\fP[\fB+w\fP\fIlon\fP/\fIlat\fP[/\fIz\fP]][\fB+v\fP\fIx0\fP/\fIy0\fP] ] [ \fB\-G\fP\fIfill\fP ] [ \fB\-Jz\fP|\fBZ\fP\fIparameters\fP ] [ \fB\-K\fP ] 
[ \fB\-L\fP[\fBf\fP][\fBx\fP]\fIlon0\fP/\fIlat0\fP[/\fIslon\fP]/\fIslat\fP/\fIlength\fP[\fBm\fP|\fBn\fP|\fBk\fP][\fB+l\fP\fIlabel\fP][\fB+j\fP\fIjust\fP][\fB+p\fP\fIpen\fP][\fB+f\fP\fIfill\fP][\fB+u\fP] ] ] [ \fB\-O\fP ] [ \fB\-P\fP ] [ \fB\-U\fP[\fIjust\fP/\fIdx\fP/\fIdy\fP/][\fBc\fP|\fIlabel\fP] ] 
[ \fB\-T\fP[\fBf\fP|\fBm\fP][\fBx\fP]\fIlon0\fP/\fIlat0\fP/\fIsize\fP[/\fIinfo\fP][\fB:\fPw,e,s,n\fB:\fP][\fB+\fP\fIgint\fP[/\fImint\fP]] ] [ \fB\-V\fP ] [ \fB\-X\fP[\fBa\fP|\fBc\fP|\fBr\fP][\fIx-shift\fP[\fBu\fP]] ] [ \fB\-Y\fP[\fBa\fP|\fBc\fP|\fBr\fP][\fIy-shift\fP[\fBu\fP]] ] [ \fB\-Z\fP\fIzlevel\fP ] [ \fB\-c\fP\fIcopies\fP ]
.SH DESCRIPTION
\fBpsbasemap\fP creates \fIPostScript\fP code that will produce a basemap.  Several map projections
are available, and the user may specify separate tickmark intervals for boundary annotation,
ticking, and [optionally] gridlines.  A simple map scale or directional rose may also be plotted.
.TP
\fB\-B\fP
Sets map boundary annotation and tickmark intervals. The format of \fItickinfo\fP is
[\fBp\fP|\fBs\fP]\fIxinfo\fP[/\fIyinfo\fP[/\fIzinfo\fP]][\fB:.\fP"Title"\fB:\fP][\fBW\fP|\fBw\fP][\fBE\fP|\fBe\fP][\fBS\fP|\fBs\fP][\fBN\fP|\fBn\fP][\fBZ\fP|\fBz\fP[\fB+\fP]].
The leading \fBp\fP [Default] or \fBs\fP selects the primary or secondary annotation information.
Each of the \fI?info\fP segments are textstrings of the form
\fIinfo\fP[\fB:\fP"Axis label"\fB:\fP][\fB:=\fP"prefix"\fB:\fP][\fB:,\fP"unit label"\fB:\fP].
The \fIinfo\fP string is made up of one or more concatenated substrings of the form
[\fBa\fP|\fBf\fP|\fBg\fP]\fIstride\fP[+-\fIphase\fP][\fIunit\fP].
The leading \fBa\fP is used to specify the annotation and major tick spacing [Default], \fBf\fP for minor tick spacing, and
\fBg\fP for gridline spacing.
\fIstride\fP is the desired stride interval.
The optional \fIphase\fP shifts the annotation interval by that amount (positive or negative).
The optional \fIunit\fP indicates the unit of the \fIstride\fP and can be any of
\fBY\fP (year, plot with 4 digits), \fBy\fP (year, plot with 2 digits), \fBO\fP (month, plot using \fBPLOT_DATE_FORMAT\fP),
\fBo\fP (month, plot with 2 digits), \fBU\fP (ISO week, plot using \fBPLOT_DATE_FORMAT\fP), \fBu\fP (ISO week, plot using 2 digits),
\fBr\fP (Gregorian week, 7-day stride from start of week \fBTIME_WEEK_START\fP), \fBK\fP (ISO weekday, plot name of day),
\fBD\fP (date, plot using \fBPLOT_DATE_FORMAT\fP), \fBd\fP (day, plot day of month 0-31 or year 1-366, via \fBPLOT_DATE_FORMAT\fP),
\fBR\fP (day, same as \fBd\fP, aligned with \fBTIME_WEEK_START\fP), \fBH\fP (hour, plot using \fBPLOT_CLOCK_FORMAT\fP),
\fBh\fP (hour, plot with 2 digits), \fBM\fP (minute, plot using \fBPLOT_CLOCK_FORMAT\fP), \fBm\fP (minute, plot with 2 digits), 
\fBC\fP (second, plot using \fBPLOT_CLOCK_FORMAT\fP), \fBc\fP (second, plot with 2 digits).  Note for geographic axes
\fBm\fP and \fBc\fP instead mean arc minutes and arc seconds.  All entities that are language-specific are under control
by \fBTIME_LANGUAGE\fP.  To specify separate x and y ticks, separate the substrings that apply to the x and y
axes with a slash [/] (If a 3-D basemap is selected with \fB\-E\fP and \fB\-Jz\fP, a third substring pertaining
to the vertical axis may be appended.)  For linear/log/power projections (\fB\-Jx\fP|\fBX\fP): Labels
for each axis can be added by surrounding them with colons (\fB:\fP).  If the first character in the
label is a period, then the label is used as plot title; if it is a comma (\fB,\fP) then the label is
appended to each annotation; if it is an equal sign (\fB=\fP) the the prefix is prepended to each annotation (start label/prefix
with - to avoid space between annotation and item); else it is the axis label.
If the label consists of more than one word, enclose the entire label in double quotes (e.g.,
\fB:\fP"my label"\fB:\fP).
If you need to use a colon (:) as part of your label you must specify it using its octal code (\\072).
.br
By default, all 4 boundaries are plotted (referred to as \fBW, E, S, N\fP).  To change the default,
append the code for only those axes you want (e.g., \fBWS\fP for standard lower-left x- and y-axis system).
Upper case (e.g., \fBW\fP) means draw axis/tickmarks AND annotate it, whereas lower case
(e.g., \fBw\fP) will only draw axis/tickmarks.  (If a 3-D basemap is selected with \fB\-E\fP and
\fB\-Jz\fP, append \fBZ\fP or \fBz\fP to control the appearance
of the vertical axis.  Append \fB+\fP to draw the outline of the cube defined by \fB\-R\fP.  Note
that for 3-D views the title, if given, will be suppressed.)
.br
For non-geographical projections:
Give negative scale (in \fB\-Jx\fP) or axis length (in \fB\-JX\fP) to change the direction
of increasing coordinates (i.e., to make the y-axis positive down).  For log10 axes:  Annotations
can be specified in one of three ways:  (1) \fIstride\fP can be 1, 2, 3, or -\fIn\fP.  Annotations will then
occur at 1, 1\-2\-5, or 1\-2\-3\-4\-...\-9, respectively; for -\fIn\fP we annotate every \fIn\fP't magnitude.
This option can also be used for
the frame and grid intervals.  (2) An \fBl\fP is appended to the \fItickinfo\fP string.
Then, log10 of the tick value is plotted at every integer log10 value.  (3) A \fBp\fP is appended
to the \fItickinfo\fP string.  Then, annotations appear as 10 raised to log10 of the tick value.
For power axes: Annotations can be specified in one of two ways:  (1) \fIstride\fP sets the
regular annotation interval.  (2) A \fBp\fP is appended to the \fItickinfo\fP string.  Then,
the annotation interval is expected to be in transformed units, but the annotation value
will be plotted as untransformed units.  E.g., if \fIstride\fP = 1 and \fIpower\fP = 0.5 (i.e., sqrt),
then equidistant annotations labeled 1\-4\-9... will appear.
.br
These GMT parameters can affect the appearance of the map boundary: \fBANNOT_MIN_ANGLE\fP, \fBANNOT_MIN_SPACING\fP,
\fBANNOT_FONT_PRIMARY\fP, \fBANNOT_FONT_SECONDARY\fP, \fBANNOT_FONT_SIZE_PRIMARY\fP, \fBANNOT_FONT_SIZE_SECONDARY\fP,
\fBANNOT_OFFSET_PRIMARY\fP, \fBANNOT_OFFSET_SECONDARY\fP, \fBBASEMAP_AXES\fP, \fBBASEMAP_FRAME_RGB\fP, \fBBASEMAP_TYPE\fP, \fBPLOT_DEGREE_FORMAT\fP,
\fBFRAME_PEN\fP, \fBFRAME_WIDTH\fP, \fBGRID_CROSS_SIZE_PRIMARY\fP, \fBGRID_PEN_PRIMARY\fP, \fBGRID_CROSS_SIZE_SECONDARY\fP,
\fBGRID_PEN_SECONDARY\fP, \fBHEADER_FONT\fP, \fBHEADER_FONT_SIZE\fP, \fBLABEL_FONT\fP, \fBLABEL_FONT_SIZE\fP, \fBLINE_STEP\fP,
\fBOBLIQUE_ANNOTATION\fP, \fBPLOT_CLOCK_FORMAT\fP, \fBPLOT_DATE_FORMAT\fP, \fBTIME_FORMAT_PRIMARY\fP, \fBTIME_FORMAT_SECONDARY\fP,
\fBTIME_LANGUAGE\fP, \fBTIME_WEEK_START\fP, \fBTICK_LENGTH\fP, \fBTICK_PEN\fP, and \fBY_AXIS_TYPE\fP; see the
\fBgmtdefaults\fP man page for details.
.TP
\fB\-J\fP
Selects the map projection. The following character determines the projection. If the
character is upper case then the argument(s) supplied as scale(s) is interpreted to be
the map width (or axis lengths), else the scale argument(s) is the map scale (see its
definition for each projection). UNIT is cm, inch, or m, depending on the \fBMEASURE_UNIT\fP
setting in \.gmtdefaults4, but this can be overridden on the command line by appending
\fBc\fP, \fBi\fP, or \fBm\fP to the \fIscale\fP or \fIwidth\fP values.  Append \fBh\fP, \fB+\fP, or \fB-\fP
to the given \fIwidth\fP if you instead want to set map height, the maximum dimension, or
the minimum dimension, respectively [Default is \fBw\fP for width].
.br
In case the central meridian is an optional parameter and it is being omitted, then the
center of the longitude range given by the \fB\-R\fP option is used. The default standard parallel
is the equator.
.br
The ellipsoid used in the map projections is user-definable by editing the \.gmtdefaults4 file
in your home directory. 73 commonly used ellipsoids and spheroids are currently
supported, and users may also specify their own custum ellipsoid parameters [Default is WGS-84].
Several GMT parameters can affect the projection: \fBELLIPSOID\fP, \fBINTERPOLANT\fP,
\fBMAP_SCALE_FACTOR\fP, and \fBMEASURE_UNIT\fP; see the \fBgmtdefaults\fP man page for details.
.br
Choose one of the following projections (The \fBE\fP or \fBC\fP after projection names
stands for Equal-Area and Conformal, respectively):
.RS
.PP
\fBCYLINDRICAL PROJECTIONS:\fP
.TP
\fB\-Jc\fP\fIlon0/lat0/scale\fP or \fB\-JC\fP\fIlon0/lat0/width\fP (Cassini).
Give projection center \fIlon0/lat0\fP and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jcyl_stere\fP/[\fIlon0/\fP[\fIlat0/\fP]]\fIscale\fP or \fB\-JCyl_stere\fP/[\fIlon0/\fP[\fIlat0/\fP]]\fIwidth\fP (Cylindrical Stereographic).
Give central meridian \fIlon0\fP (optional), standard parallel \fIlat0\fP (optional), and \fIscale\fP along parallel (\fB1:\fP\fIxxxx\fP or UNIT/degree).
The standard parallel is typically one of these (but can be any value):
.RS
.RS
66.159467 - Miller's modified Gall
.br
55 - Kamenetskiy's First
.br
45 - Gall's Stereographic
.br
30 - Bolshoi Sovietskii Atlas Mira or Kamenetskiy's Second
.br
0 - Braun's Cylindrical
.RE
.RE
.TP
\fB\-Jj\fP[\fIlon0/\fP]\fIscale\fP or \fB\-JJ\fP[\fIlon0/\fP]\fIwidth\fP (Miller Cylindrical Projection).
Give the central meridian \fIlon0\fP (optional) and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jm\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIscale\fP or \fB\-JM\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIwidth\fP
Give central meridian \fIlon0\fP (optional), standard parallel \fIlat0\fP (optional), and \fIscale\fP along parallel (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jo\fP\fIparameters\fP (Oblique Mercator \fB[C]\fP).
Typically used with \fB\-R\fP<...>\fBr\fP, otherwise region is in oblique coordinates. Specify one of:
.RS
.TP
\fB\-Jo\fP[\fBa\fP]\fIlon0/lat0/azimuth/scale\fP or \fB\-JO\fP[\fBa\fP]\fIlon0/lat0/azimuth/width\fP
Set projection center \fIlon0/lat0\fP, \fIazimuth\fP of oblique equator, and \fIscale\fP.
.TP
\fB\-Jo\fP[\fBb\fP]\fIlon0/lat0/lon1/lat1/scale\fP or \fB\-JO\fP[\fBb\fP]\fIlon0/lat0/lon1/lat1/scale\fP
Set projection center \fIlon0/lat0\fP, another point on the oblique equator \fIlon1/lat1\fP, and \fIscale\fP.
.TP
\fB\-Joc\fP\fIlon0/lat0/lonp/latp/scale\fP or \fB\-JOc\fP\fIlon0/lat0/lonp/latp/scale\fP
Set projection center \fIlon0/lat0\fP, pole of oblique projection \fIlonp/latp\fP, and \fIscale\fP.
.PP
Give \fIscale\fP along oblique equator (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.RE
.TP
\fB\-Jq\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIscale\fP or \fB\-JQ\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIwidth\fP (Cylindrical Equidistant).
Give the central meridian \fIlon0\fP (optional), standard parallel \fIlat0\fP (optional), and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
The standard parallel is typically one of these (but can be any value):
.RS
.RS
61.7 - Grafarend and Niermann, minimum linear distortion
.br
50.5 - Ronald Miller Equirectangular
.br
43.5 - Ronald Miller, minimum continental distortion
.br
42 - Grafarend and Niermann
.br
37.5 - Ronald Miller, minimum overall distortion
.br
0 - Plate Carree, Simple Cylindrical, Plain/Plane Chart
.br
.RE
.RE
.TP
\fB\-Jt\fP\fIlon0/\fP[\fIlat0/\fP]\fIscale\fP or \fB\-JT\fP\fIlon0/\fP[\fIlat0/\fP]\fIwidth\fP
Give the central meridian \fIlon0\fP, central parallel \fIlat0\fP (optional), and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Ju\fP\fIzone/scale\fP or \fB\-JU\fP\fIzone/width\fP (UTM - Universal Transverse Mercator \fB[C]\fP).
Give the UTM zone (A,B,1-60[C-X],Y,Z)) and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.br
Zones: If C-X not given, prepend - or + to enforce southern or northern hemisphere conventions [northern if south > 0].
.TP
\fB\-Jy\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIscale\fP or \fB\-JY\fP[\fIlon0/\fP[\fIlat0/\fP]]\fIwidth\fP (Cylindrical Equal-Area \fB[E]\fP).
Give the central meridian \fIlon0\fP (optional), standard parallel \fIlat0\fP (optional), and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
The standard parallel is typically one of these (but can be any value):
.RS
.RS
50 - Balthasart
.br
45 - Gall-Peters
.br
37.0666 - Caster
.br
37.4 - Trystan Edwards
.br
37.5 - Hobo-Dyer
.br
30 - Behrman
.br
0 - Lambert (default)
.RE
.RE
.PP
\fBCONIC PROJECTIONS:\fP
.TP
\fB\-Jb\fP\fIlon0/lat0/lat1/lat2/scale\fP or \fB\-JB\fP\fIlon0/lat0/lat1/lat2/width\fP (Albers \fB[E]\fP).
Give projection center \fIlon0/lat0\fP, two standard parallels \fIlat1/lat2\fP, and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jd\fP\fIlon0/lat0/lat1/lat2/scale\fP or \fB\-JD\fP\fIlon0/lat0/lat1/lat2/width\fP (Conic Equidistant)
Give projection center \fIlon0/lat0\fP, two standard parallels \fIlat1/lat2\fP, and \fIscale\fP (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jl\fP\fIlon0/lat0/lat1/lat2/scale\fP or \fB\-JL\fP\fIlon0/lat0/lat1/lat2/width\fP (Lambert \fB[C]\fP)
Give origin \fIlon0/lat0\fP, two standard parallels \fIlat1/lat2\fP, and \fIscale\fP along these (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.TP
\fB\-Jpoly\fP/[\fIlon0/\fP[\fIlat0/\fP]]\fIscale\fP or \fB\-JPoly\fP/[\fIlon0/\fP[\fIlat0/\fP]]\fIwidth\fP ((American) Polyconic).
Give the central meridian \fIlon0\fP (optional), reference parallel \fIlat0\fP (optional, default = equator),
and \fIscale\fP along central meridian (\fB1:\fP\fIxxxx\fP or UNIT/degree).
.PP
\fBAZIMUTHAL PROJECTIONS:\fP
.sp
Except for polar aspects, \fB\-R\fP w/e/s/n will be reset to \fB\-Rg\fP.  Use \fB\-R\fP<...>\fBr\fP for smaller regions.
.TP
\fB\-Ja\fP\fIlon0/lat0\fP[\fI/horizon\fP]\fI/scale\fP or \fB\-JA\fP\fIlon0/lat0\fP[\fI/horizon\fP]\fI/width\fP (Lambert \fB[E]\fP).
\fIlon0/lat0\fP specifies the projection center.
\fIhorizon\fP specifies the max distance from projection center (in degrees, <= 180, default 90).
Give \fIscale\fP as \fB1:\fP\fIxxxx\fP or \fIradius/lat\fP, where \fIradius\fP is distance
in UNIT from origin to the oblique latitude \fIlat\fP.
.TP

gmt-4.5.14/ChangeLog
-----------------------------------------------------------------------------------------
	$Id: ChangeLog 10331 2015-10-29 19:41:56Z pwessel $
-----------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------
2015-11-01	pwessel
	* GMT 4.5.14 finalized and released
-----------------------------------------------------------------------------------------
2015-10-29	pwessel
	* meca/psmeca.c:	Enforce rakes and strikes to be in -180/+180 range.
				Issue #799, for gmt4 only as not a problem in gmt5.

2015-07-28	pwessel
	* meca/psmeca.c:	Be able to read any value x and y coordinates

2015-06-24	pwessel
	* gmt_support.c:	Fix bug in jump near Dateline in gmt_inonout_sphpol_count,
				affecting inside/outside spherical cases, reported in message-2219

2015-03-18	pwessel
	* share/cpt/GMT_wysiwyg.cpt:	Fix typo in blue for high end [Issue #689].

2015-01-16	pwessel
	* gmt_plot.c:		Fix issue # 662 with graph arrows for negative scales.

2015-01-10	pwessel
	* gmt_plot.c:		Flipping the n/s sides array when lon and -y axes
				fixes issue #520

2015-01-01	pwessel
	* configure.ac:		Moved to 4.5.14 and back to svn release only.
-----------------------------------------------------------------------------------------
2015-01-01	pwessel
	* GMT 4.5.13 finalized and released [GSHHG 2.3.4 was released on Jan 1 as well]
-----------------------------------------------------------------------------------------
2014-12-28	pwessel
	* gmt_map.c:		Needed separate GMT_cartesian_overlap for non-geographic
				projections.  Fixes issue # 657.

2014-09-29	pwessel
	* gmt_plot.c:		Left a PSL variable undefined when grdimage plotted a grid
				with x = longitude and y = Cartesian.

2014-09-09	pwessel
	* grdfilter.c:		Fix y-shifts when -I sets smaller output.  Issue #616.
	* gmt_bcr.c:		Fix problem with points slightly outside causing an
				access violation; see issue #620.

2014-08-29	pwessel
	* sample1d.c:		Fix resampling for decreasing t values.  Issue #608.

2014-08-05	pwessel
	* gmt_map.c:		Could run into trouble clipping polygons for non-periodic
				maps.  Added more checks in GMT_wesn_clip.

2014-07-31	pwessel
	* gmt_map.c:		Rect clipping with -R<w/s/e/n>r could introduce stray lines
				when dataset has features about 180 degrees away from
				the user's area. See issue # 594.

2014-07-30	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.2 (released August 1, 2014)
	* psxy[z].c:		Fixed wrap-around issue in fault lines.

2014-07-27	pwessel
	* gmtselect.c:		The -Z option needed to pass records with NaNs.

2014-07-19	pwessel
	* x2sys/x2sys_solve.c:	Fix problems that arise when a track has less crossovers
				than model parameters.
	* gmtmath.c:		Fix issue #123 with files and -C.

2014-07-01	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.1 (released July 1, 2014)
	
2014-06-28	pwessel
	* mgd77/mgd77sniffer.c:	Needed to check that grid values used in decimate function
	 			also were within reasonable range.

2014-05-26	pwessel
	* gmt_map.c:		The map region clip path for -JE/e was hardwired to a full
				circle and did not care about -R setting.
				When -R...r and -Je/E was given the search for enclosing
				geographic boundaries failed when either the S or N pole
				was the projection's antipole.
2014-05-25	pwessel
	* grdmath.c:		Fix issue #566 bug in grdmath's KM2DEG operator.

2014-03-19	pwessel
	* gmtselect.c:		The -Fpolygon option might miss points because the io-machinery
				would sometimes set the min/max longitudes found incorrectly.

2014-03-14	pwessel
	* gmt_shore.c:		Give more error feedback if nc_open fails due to old library.

2014-03-01	pwessel
	* configure.ac:		Moved to 4.5.13 and back to svn release only.

-----------------------------------------------------------------------------------------
2014-03-01	pwessel
	* GMT 4.5.12 finalized and released [GSHHG 2.3.0 was released on Feb 1]
-----------------------------------------------------------------------------------------
2014-02-17	fwobbe
	* gmt_init.c etc.:	File locking -- if enabled via -DFLOCK -- is
				non-compulsory now.  This means if fcntl fails for some reason,
				GMT continues to access the file.  Hence, it is recommended to
				configure GMT with --enable-flock even if fcntl might be buggy
				on some ancient NFS shares.

2014-02-17	pwessel
	* gmt_map.c:		Size for memory allocation cast the product of nx * ny to
				size_t but overflow had already happen as calculation was
				done using ints, then cast to size_t. Must cast nx and ny
				separately.
2014-02-10	jluis
	* misc/gmt2kml.c:	Make sure the default style IDs are unique for each process.

2014-02-07	jluis
	* ps2raster.c:		Make sure that images in KML files do not carry a directory name

2014-02-03	pwessel
	* pslib.c:		Auto-widen paragraph width to at least fit widest word.

2014-02-01	pwessel
	* configure.ac:		Since GSHHG 2.3.0 was released today, make that the default

2014-01-26	pwessel
	* gmt_grdio.c:		Building the grid command string could exceed its length if
				the argument became exactly the max length of the array; then
				a final extra character \0 would be added and exceed array length.
				Thanks to Joachim Saul for pointing it out.

2014-01-22	pwessel
	* pscoast.c:		The processing of -I -N levels and pens could get out of sync
				so that the wrong pen was used for the specified feature.

2014-01-16	remko
	* pslib.c:		Take superscripts and subscripts into account when determining
				dimensions of text box.

2014-01-10	pwessel
	* grdmath.c:		Had to let SDIST return km instead of degrees since it cannot
				do degrees for geodesics.  By doing km it is now aligned with
				what PDIST and LDIST returns for geographic data, and matches
				what GMT5 does.  Some scripts had to be modified to handle the
				new behavior.  If you use SDIST you may wish to check your usage.
				Also fixed bug in SDIST in that spherical selected geodesics

gmt-4.5.14/src/psbasemap.c
/*--------------------------------------------------------------------
 *	$Id: psbasemap.c 10289 2014-12-28 21:17:06Z pwessel $
 *
 *	Copyright (c) 1991-2015 by P. Wessel and W. H. F. Smith
 *	See LICENSE.TXT file for copying and redistribution conditions.
 *
 *	This program is free software; you can redistribute it and/or modify
 *	it under the terms of the GNU General Public License as published by
 *	the Free Software Foundation; version 2 or any later version.
 *
 *	This program is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU General Public License for more details.
 *
 *	Contact info: gmt.soest.hawaii.edu
 *--------------------------------------------------------------------*/
/*
 * psbasemap plots out a basemap for the given area using the specified map
 * projection.
 *
 * Author:	Paul Wessel
 * Date:	08-JUL-2000
 * Version:	4
 *
 */
 
#include "gmt.h"
#include "pslib.h"

struct PSBASEMAP_CTRL {
	struct E {	/* -Eazim/elev */
		GMT_LONG active;
		double azimuth, elevation;
	} E;
	struct G {	/* -G<fill> */
		GMT_LONG active;
		struct GMT_FILL fill;
	} G;
	struct L {	/* -L */
		GMT_LONG active;
		struct GMT_MAP_SCALE item;
	} L;
	struct T {	/* -L */
		GMT_LONG active;
		struct GMT_MAP_ROSE item;
	} T;
	struct Z {	/* -Z<z_level> */
		GMT_LONG active;
		double level;
	} Z;
};

int main (int argc, char **argv)
{
	GMT_LONG i;

	GMT_LONG error = FALSE;

	double w = 0.0, e = 0.0, s = 0.0, n = 0.0;

	struct PSBASEMAP_CTRL *Ctrl = NULL;

	void *New_psbasemap_Ctrl (), Free_psbasemap_Ctrl (struct PSBASEMAP_CTRL *C);
	
	argc = (int)GMT_begin (argc, argv);

	Ctrl = (struct PSBASEMAP_CTRL *)New_psbasemap_Ctrl ();	/* Allocate and initialize a new control structure */

	for (i = 1; i < argc; i++) {
		if (argv[i][0] == '-') {
			switch (argv[i][1]) {

				/* Common parameters */

				case 'B':
				case 'J':
				case 'K':
				case 'O':
				case 'P':
				case 'R':
				case 'U':
				case 'V':
				case 'X':
				case 'x':
				case 'Y':
				case 'y':
				case 'c':
				case '\0':
					error += GMT_parse_common_options (argv[i], &w, &e, &s, &n);
					break;

				/* Supplemental options */

				case 'E':
					Ctrl->E.active = TRUE;
					error += GMT_get_proj3D (&argv[i][2], &Ctrl->E.azimuth, &Ctrl->E.elevation);
					break;

				case 'G':
					Ctrl->G.active = TRUE;
					if (GMT_getfill (&argv[i][2], &Ctrl->G.fill)) {
						GMT_fill_syntax ('G', " ");
						error++;
					}
					break;

				case 'L':
					Ctrl->L.active = TRUE;
					error += GMT_getscale (&argv[i][2], &Ctrl->L.item);
					break;

				case 'T':
					Ctrl->T.active = TRUE;
					error += GMT_getrose (&argv[i][2], &Ctrl->T.item);
					break;

				case 'Z':
					if (argv[i][2]) {
						Ctrl->Z.level = atof (&argv[i][2]);
						Ctrl->Z.active = TRUE;
					}
					else {
						error++;
						fprintf (stderr, "%s: GMT SYNTAX ERROR -Z:  Must append a z-value\n", GMT_program);
					}
					break;

				/* Illegal options */

				default:
					error = TRUE;
					GMT_default_error (argv[i][1]);
					break;
			}
		}
		else
			fprintf (stderr, "%s: Warning: Ignoring filename %s\n", GMT_program, argv[i]);
	}

	if (GMT_give_synopsis_and_exit || argc == 1) {
		fprintf (stderr,"psbasemap %s - To plot PostScript basemaps\n\n", GMT_VERSION);
		fprintf (stderr, "usage: psbasemap %s %s %s [%s] [-G<fill>]\n", GMT_B_OPT, GMT_J_OPT, GMT_Rgeoz_OPT, GMT_E_OPT);
		fprintf (stderr, "\t[-K] [%s] [%s]\n", GMT_Jz_OPT, GMT_SCALE);
		fprintf (stderr, "\t[-O] [-P] [%s] [%s] [-V]\n", GMT_TROSE, GMT_U_OPT);
		fprintf (stderr, "\t[%s] [%s] [-Z<zlevel>] [%s]\n\n", GMT_X_OPT, GMT_Y_OPT, GMT_c_OPT);

		if (GMT_give_synopsis_and_exit) exit (EXIT_FAILURE);

		GMT_explain_option ('B');
		GMT_explain_option ('J');
		GMT_explain_option ('Z');
		GMT_explain_option ('R');
		fprintf (stderr, "\n\tOPTIONS:\n");
		GMT_explain_option ('E');
		GMT_fill_syntax ('G', "Select fill inside of basemap.");
		GMT_explain_option ('K');
		GMT_mapscale_syntax ('L', "Draws a simple map scale centered on <lon0>/<lat0>.");
		GMT_explain_option ('O');
		GMT_explain_option ('P');
		GMT_maprose_syntax ('T', "Draws a north-pointing map rose centered on <lon0>/<lat0>.");
		GMT_explain_option ('U');
		GMT_explain_option ('V');
		GMT_explain_option ('X');
		fprintf (stderr, "\t-Z For 3-D plots: Set the z-level of map [Default is at bottom of z-axis].\n");
		GMT_explain_option ('c');
		GMT_explain_option ('.');
		exit (EXIT_FAILURE);
	}

	if (!project_info.region_supplied) {
		fprintf (stderr, "%s: GMT SYNTAX ERROR:  Must specify -R option\n", GMT_program);
		error++;
	}
	if (!(frame_info.plot || Ctrl->L.active || Ctrl->T.active || Ctrl->G.active)) {
		fprintf (stderr, "%s: GMT SYNTAX ERROR:  Must specify at least one of -B, -G, -L, -T\n", GMT_program);
		error++;
	}
	if (Ctrl->E.elevation <= 0.0 || Ctrl->E.elevation > 90.0) {
		fprintf (stderr, "%s: GMT SYNTAX ERROR -E option:  Elevation must be in 0-90 range\n", GMT_program);
		error++;
	}
	if (Ctrl->L.active && ! GMT_IS_MAPPING) {
		fprintf (stderr, "%s: GMT SYNTAX ERROR:  -L applies to geographical data only\n", GMT_program);
		error++;
	}
	if (error) exit (EXIT_FAILURE);

	z_project.view_azimuth = Ctrl->E.azimuth;
	z_project.view_elevation = Ctrl->E.elevation;

	if (gmtdefs.verbose) fprintf (stderr, "psbasemap: Constructing basemap\n");

	GMT_err_fail (GMT_map_setup (w, e, s, n), "");

	GMT_plotinit (argc, argv);

	if (project_info.three_D) ps_transrotate (-z_project.xmin, -z_project.ymin, 0.0);

	if (Ctrl->G.active) {
		double *x, *y;
		GMT_LONG np;
		GMT_LONG donut;
		np = GMT_map_clip_path (&x, &y, &donut);
		GMT_fill (x, y, (1 + donut) * np, &Ctrl->G.fill, FALSE);
		GMT_free ((void *)x);
		GMT_free ((void *)y);
	}

	if (Ctrl->Z.active) project_info.z_level = Ctrl->Z.level;

	GMT_map_basemap ();

	if (Ctrl->L.active) GMT_draw_map_scale (&Ctrl->L.item);

	if (Ctrl->T.active) GMT_draw_map_rose (&Ctrl->T.item);

	if (project_info.three_D) ps_rotatetrans (z_project.xmin, z_project.ymin, 0.0);
	
	GMT_plotend ();

	Free_psbasemap_Ctrl (Ctrl);	/* Deallocate control structure */

	GMT_end (argc, argv);

	exit (EXIT_SUCCESS);
}

void *New_psbasemap_Ctrl () {	/* Allocate and initialize a new control structure */
	struct PSBASEMAP_CTRL *C;
	
	C = (struct PSBASEMAP_CTRL *) GMT_memory (VNULL, (size_t)1, sizeof (struct PSBASEMAP_CTRL), "New_psbasemap_Ctrl");
	
	/* Initialize values whose defaults are not 0/FALSE/NULL */
	C->E.azimuth = 180.0;
	C->E.elevation = 90.0;
	GMT_init_fill (&C->G.fill, -1, -1, -1);
	memset ((void *)&C->L.item, 0, sizeof (struct GMT_MAP_SCALE));
	memset ((void *)&C->T.item, 0, sizeof (struct GMT_MAP_ROSE));
		
	return ((void *)C);
}

void Free_psbasemap_Ctrl (struct PSBASEMAP_CTRL *C) {	/* Deallocate control structure */
	GMT_free ((void *)C);	
}
